### PR TITLE
Add maxlength on input components

### PR DESCRIPTION
**Pull request by chromaticWaster**
### Description:

The request for this component was to add a max length property to specific input components 

- Omni Text Field
- Omni Email Field
- Omni Search Field
- Omni Number Field
- Omni Password Field

Specific doc site examples were added for each component along with the relevant max-length test.

### All Submissions:

* [x] I have followed the [contribution guidelines](https://github.com/capitec/omni-components/blob/develop/CONTRIBUTING.md) for this project.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

File: src/text-field/TextField.stories.ts [hunk 1]
```diff
@@ -17,12 +18,17 @@ import { assignToSlot, ComponentStoryFormat, getSourceFromLit } from '../utils/S
...
+    maxLength: number;
+}
+
+export const Interactive: ComponentStoryFormat<Args> = {
+    render: (args: Args) => html`
     <omni-text-field
       data-testid="test-text-field"
       label="${ifNotEmpty(args.label)}"
       value="${args.value}"
+      max-length=${args.maxLength}
```
Comment (BOTLANNER) [inline src/text-field/TextField.stories.ts:31]: Should this not be ifNotEmpty?

File: src/email-field/EmailField.stories.ts [hunk 2]
```diff
@@ -16,13 +17,17 @@ import { ifNotEmpty } from '../utils/Directives.js';
...
 
-export const Interactive: ComponentStoryFormat<BaseArgs> = {
-    render: (args: BaseArgs) => html`
+export const Interactive: ComponentStoryFormat<Args> = {
+    render: (args: Args) => html`
     <omni-email-field
       data-testid="test-email-field"
       label="${ifNotEmpty(args.label)}"
       value="${args.value}"
+      max-length=${args.maxLength}
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/email-field/EmailField.stories.ts]: `ifDefined` implemented

File: src/number-field/NumberField.stories.ts [hunk 3]
```diff
@@ -16,13 +17,17 @@ import { ifNotEmpty } from '../utils/Directives.js';
...
 
-export const Interactive: ComponentStoryFormat<BaseArgs> = {
-    render: (args: BaseArgs) => html`
+export const Interactive: ComponentStoryFormat<Args> = {
+    render: (args: Args) => html`
     <omni-number-field
       data-testid="test-number-field"
       label="${ifNotEmpty(args.label)}"
       value="${args.value}"
+      max-length=${args.maxLength}
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/number-field/NumberField.stories.ts]: `ifDefined` implemented

File: src/password-field/PasswordField.stories.ts [hunk 4]
```diff
@@ -31,6 +33,7 @@ export const Interactive: ComponentStoryFormat<Args> = {
       data-testid="test-password-field"
       label="${ifNotEmpty(args.label)}"
       value="${args.value}"
+      max-length=${args.maxLength}
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/password-field/PasswordField.stories.ts]: `ifDefined` implemented

File: src/search-field/SearchField.stories.ts [hunk 5]
```diff
@@ -16,12 +17,17 @@ import { ifNotEmpty } from '../utils/Directives.js';
...
+    maxLength: number;
+}
+
+export const Interactive: ComponentStoryFormat<Args> = {
+    render: (args: Args) => html`
         <omni-search-field
             data-testid="test-search-field"
             label="${ifNotEmpty(args.label)}"
             value="${args.value}"
+            max-length=${args.maxLength}
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/search-field/SearchField.stories.ts]: `ifDefined` implemented

File: src/text-field/TextField.stories.ts (hunk 1 above)
Comment (chromaticWaster) [reply src/text-field/TextField.stories.ts]: `ifDefined` implemented

**Review by stefan505 — APPROVED** [review]

**Pull request merged**